src/pip/_internal/pep425tags.py

```diff
@@ -263,7 +280,8 @@ def get_supported(versions=None, noarch=False, platform=None,
 
     if not noarch:
         arch = platform or get_platform()
-        if arch.startswith('macosx'):
+        arch_prefix, arch_sep, arch_suffix = arch.partition('_')
+        if arch_prefix == 'macosx':
```

Comment:
I'm not sure how the `macosx` platform tags are actually structured, so it would be safer to keep the `startswith` for this case.

Reply:
Agreed, I've reverted this line to use `startswith`.